### DISCUSSION TITLE
How to join (and select) multiple tables?

### DISCUSSION THREAD

**Discussion by smessmer** [SeaQL/sea-orm#1975] [Q&A]
I have a table A. Each row has exactly one corresponding row in exactly one of the tables B, C or D. I'm using this to implement an inheritance-like pattern where B, C, D are different entities and A is a common table containing common shared properties, e.g. their id. I now want to query such an entity by id but don't know its type. So I want to write a query that joins all these tables together and gives me the data I need. Something like
```sql
SELECT
  *
FROM A
LEFT JOIN B ON A.id = B.id
LEFT JOIN C ON A.id = C.id
LEFT JOIN D ON A.id = D.id
WHERE A.id = ...
```

I've seen that sea-orm has good support for one left join, by doing:
```rust
let rows: Vec<(ModelA, Option<ModelB>)> =
  EntityA::find()
    .find_also_related(EntityB)
    .all(db).await
```

Is there a way to do this for multiple joins? I tried:
```rust
let rows: Vec<(ModelA, Option<ModelB>, Option<ModelC>, Option<ModelD>)> =
  EntityA::find()
    .find_also_related(EntityB)
    .find_also_related(EntityC)
    .find_also_related(EntityD)
    .all(db).await
```

but that doesn't compile because `find_also_related` returns a `SelectTwo` instance that doesn't allow further calls to `find_also_related`.

Other combinations I tried also didn't work. I did figure out how to chain multiple calls to `QuerySelect::join`, but then I didn't find a way to add those joined table rows to the select criteria. It would do the join but it would not return its results.

Is there a way to do this? Ideally in a way that it returns Model objects so I don't have to manually specify each column?

**Comment by YiNNx** [comment]
Hi, maybe this's what you need: https://www.sea-ql.org/SeaORM/docs/relation/chained-relations/
